src/lib/userStorage/useUserProfile.js

```diff
@@ -0,0 +1,9 @@
+// an useGunProfile hook implementation for the RealmDB
+//import logger from '../logger/pino-logger'
+//import userStorage from './UserStorage'
+
+//const log = logger.child({ from: 'useUserProfile' })
+
+const useUserProfile = (identifier, fields = ['fullName', 'smallAvatar']) => {}
```

Comment:
here you need to rewrite useGunProfile hook via realmDB implementation
add `profile` state var and fill / update it in the on mount useeffect:

1. read user's profile by address by its address or pkey (see changes requested for realmdb class) set it to the state var
2. then subscribe to profile changes, update state var
3. return cancel hook unwatching changes

Reply:
```javascript
onst useUserProfile = (identifier, fields = ['fullName', 'smallAvatar']) => {
  const [profile, setProfile] = useState(null)

  const updateProfile = useCallback(
    profile => setProfile(mapValues(pick(profile, fields), 'display')),
    [setProfile, fields],
  )

  useEffect(() => {
    let hookCancelled = false

    userStorage.db.getProfile(identifier).then(async profile => {
      if (hookCancelled) return
      updateProfile(profile)

      for await (const { fullDocument } of userStorage.db.getProfile(identifier)) {
        if (hookCancelled) break
        updateProfile(fullDocument)
      }
    })

    return () => void (hookCancelled = true)
  }, [identifier, updateProfile])
}

export default useUserProfile
```

Reply:
this is very raw implementation, they should be appended with logging, debug etc (pls look at useGUNProfile)

Reply:
and of course tested if it actually works (especially if the cancel hook dismisses the watch)

Reply:
no need to listen for changes